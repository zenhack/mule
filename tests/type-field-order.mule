#XFAIL: type member
#
# This test checks that the order of the fields doesn't matter wrt. types;
# it should be OK to have a type field after a value field that uses the
# type.
{
    , x : t = 4
    , type t = int
}
