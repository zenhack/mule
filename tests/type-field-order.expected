
Desugared:

	(letrec (types ((t (rec t int)))) (values ((x (: (Integer 4) t))))
	 (((.= x) (.type= t {} t)) x))

inferred type:

	(record (types ((t int))) (values ((x int))))

Runtime term:

	(LetRec
	 (((0 (Const (Integer 4))))
	  (App
	   ((App
	     ((Lam
	       (0 () (Lam (1 () (Update (old (Var 1)) (label x) (field (Var 0)))))))
	      (Record ())))
	    (Var 0)))))

Evaluated:

	(Record ((x (Const (Integer 4)))))
